package-builder/flox-build.mk

```diff
@@ -195,23 +206,45 @@ define CLEAN_result_link_template =
     $(if $(_storePath),clean_result_storepath/$(1))
 endef
 
+# The following env vars need to be passed from the outer "develop" environment
+# to the inner "wrapper" environment in the in-situ build mode in support of
+# the tools and compilers found in the outer environment, and for Flox to
+# otherwise function properly.
+ALLOW_OUTER_ENV_VARS := FLOX_RUNTIME_DIR HOME PATH \
+  $(filter NIX_CFLAGS% NIX_CC%,$(.VARIABLES))
+
 # The following template renders targets for the in-situ build mode.
 define BUILD_local_template =
   $(eval _virtualSandbox = $(filter-out null off,$(_sandbox)))
 
+  # Set temp outpath of same strlen as eventual package storePath using the
+  # 32-char hash previously derived from the package name, current working
+  # directory and FLOX_ENV.
+  $(eval _out = /tmp/store_$($(_pvarname)_hash)-$(_name))
+
+  # Make sure to invoke the build script in a nested activation of both the
+  # "develop" and "wrapper" environments, and that the wrapper environment is
+  # the "inner" activation preferred for sourcing commands, libraries, etc.
+  # Also blat all env variables set by the outer activation to avoid
+  # including the "develop" environment in the build closure.
   .INTERMEDIATE: $(_pname)_local_build
   $(_pname)_local_build: $($(_pvarname)_buildScript)
 	@# $(if $(FLOX_INTERPRETER),,$$(error FLOX_INTERPRETER not defined))
 	@echo "Building $(_name) in local mode"
-	$(if $(_virtualSandbox),$(PRELOAD_ARGS) FLOX_SRC_DIR=$$$$($(_pwd)) FLOX_VIRTUAL_SANDBOX=$(_sandbox)) \
-	MAKEFLAGS= out=$(_out) $(FLOX_INTERPRETER)/activate --turbo -- $(_bash) -e $($(_pvarname)_buildScript)
+	@$(_rm) -rf $(_out)
+	$(if $(_virtualSandbox),$(PRELOAD_VARS) FLOX_SRC_DIR=$$$$($(_pwd)) FLOX_VIRTUAL_SANDBOX=$(_sandbox)) \
+	$(FLOX_INTERPRETER)/activate --env $(FLOX_ENV) --mode dev --turbo -- \
+	  $(_env) -i out=$(_out) $(foreach i,$(ALLOW_OUTER_ENV_VARS),$(i)="$$$$$(i)") \
```

Comment:
I wonder if this should be `PREFER_OUTER_ENV_VARS` since they eplicitly override what is defined in the wrapper-env, no?

Reply:
yeah makes sense, thanks for clarifying
Although that still sounds like you can get somewhat inconsistent variable settings as a subset might be reset by the inner env